frontend/src/features/properties/filter/PropertyFilter.MOTI.test.tsx

```diff
@@ -0,0 +1,327 @@
+      isDisabled: false,
+      type: API.PROPERTY_CLASSIFICATION_CODE_SET_NAME,
+    },
+    {
+      name: 'Demolished',
+      id: '5',
+      isDisabled: false,
+      type: API.PROPERTY_CLASSIFICATION_CODE_SET_NAME,
+    },
+    {
+      name: 'Subdivided',
+      id: '6',
+      isDisabled: false,
+      type: API.PROPERTY_CLASSIFICATION_CODE_SET_NAME,
+    },
+  ] as ILookupCode[],
+};
+
+const getStore = (filter: any) =>
+  mockStore({
+    [filterSlice.name]: filter,
+    [propertyNameSlice.name]: ['test'],
+    [lookupCodesSlice.name]: lCodes,
+  });
+
+const defaultFilter: IPropertyFilter = {
+  searchBy: 'name',
+  pid: '',
+  address: '',
+  administrativeArea: '',
+  propertyType: '',
+  projectNumber: '',
+  agencies: '',
+  classificationId: '',
+  minLotSize: '',
+  maxLotSize: '',
+  rentableArea: '',
+  name: '',
+};
+
+const getUiElement = (filter: IPropertyFilter, showAllAgencySelect = true) => (
+  <TenantProvider>
+    <Provider store={getStore(filter)}>
+      <Router history={history}>
+        <PropertyFilter
+          defaultFilter={filter}
+          agencyLookupCodes={MOCK.AGENCIES}
+          adminAreaLookupCodes={MOCK.ADMINISTRATIVEAREAS}
+          onChange={onFilterChange}
+          showAllAgencySelect={showAllAgencySelect}
+        />
+      </Router>
+    </Provider>
+  </TenantProvider>
+);
+
+describe('[ MOTI ] MapFilterBar', () => {
+  mockedAxios.get.mockImplementationOnce(() => Promise.resolve({}));
+
+  beforeEach(() => {
+    process.env.REACT_APP_TENANT = 'MOTI';
+    history = createMemoryHistory();
+    mockKeycloak([]);
+  });
+
+  afterEach(() => {
+    delete process.env.REACT_APP_TENANT;
+    cleanup();
+  });
+
+  it('renders correctly', () => {
+    mockKeycloak(['property-view']);
+    // Capture any changes
+    const { container } = render(getUiElement(defaultFilter));
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  it('renders only my agencies if showAllAgencies not set', () => {
+    mockKeycloak(['property-view']);
+    // Capture any changes
+    const { getByPlaceholderText } = render(getUiElement(defaultFilter, false));
+    expect(getByPlaceholderText('My Agencies')).toBeVisible();
+  });
+
+  it('submits correct values', async () => {
+    // Arrange
+    mockKeycloak(['admin-properties']);
+
+    const { container } = render(getUiElement(defaultFilter));
+    const classificationId = container.querySelector('select[name="classificationId"]');
+    const submit = container.querySelector('button[type="submit"]');
+
+    // Act
+    // Enter values on the form fields, then click the Search button
+    await fillInput(container, 'administrativeArea', 'Victoria', 'typeahead');
+
+    await wait(() => {
+      fireEvent.change(classificationId!, {
+        target: {
+          value: '0',
+        },
+      });
+    });
+
+    await wait(() => {
+      fireEvent.click(submit!);
+    });
+
+    // Assert
+    expect(onFilterChange).toBeCalledWith<[IPropertyFilter]>({
```

Comment:
I don't see onFilterChange being reset in any of the before or after blocks.

Reply:
This is straight copy pasta from the original file so we should file a tech debt item to go over tests looking for what you mentioned